### PR TITLE
feat: Override entry_points in multitest helpers

### PR DESCRIPTION
**Pull request by jawoznia**
Part of #189

### CODE REVIEW

File: sylvia-derive/src/multitest.rs [hunk 1]
```diff
@@ -696,32 +702,64 @@ impl<'a> MultitestHelpers<'a> {
...
-        } else {
-            quote! {
-                #sylvia ::anyhow::bail!("reply not implemented for contract")
-            }
+        let reply_body = match override_entry_points.get_entry_point(MsgType::Reply) {
+            Some(entry_point) => entry_point.emit_multitest_dispatch(),
+            None => self
+                .reply
+                .as_ref()
+                .map(|reply| {
```
Comment (hashedone) [inline sylvia-derive/src/multitest.rs:753]: I don't think that you need those external curly braces, so if it is not becasue of some formatting issue I would get rid of them.

**Review by hashedone — COMMENTED** [review]
In general looks fine, but I am not sure what happens if you overwrite the entry point which has messages defined on the contract?

What should happen is that it should use the overwritten entry point, end every definition of message related to the entry point should emit a warning that entry point is overwritten.

What I would like at least is the test case for this situation - eg contract with execution overwritten and some `#[msg(exec)]` things, and then some MT verifying the overwritten one is executed.

File: sylvia-derive/src/multitest.rs (hunk 1 above)
Comment (jawoznia) [reply sylvia-derive/src/multitest.rs:753]: It is because of the formatting unfortunately

**Comment by jawoznia** [comment]
> In general looks fine, but I am not sure what happens if you overwrite the entry point which has messages defined on the contract?
> 
> What should happen is that it should use the overwritten entry point, end every definition of message related to the entry point should emit a warning that entry point is overwritten.
> 
> What I would like at least is the test case for this situation - eg contract with execution overwritten and some `#[msg(exec)]` things, and then some MT verifying the overwritten one is executed.

Done, but I took another approach and let the user to decide. User can still use generated messages but need to provide own dispatch logic.

**Review by hashedone — APPROVED** [review]
Ok, at least I see it tested now. We will think about any warning later if we would see it becomes an issue.

Ready to merge

**Pull request merged**